server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/MIMEMessageConverter.java

```diff
@@ -220,6 +216,46 @@ private Multipart createMultipart(CreationMessage newMessage, ImmutableList<Mess
         }
     }
 
+    private Multipart createMultipartWithAttachments(CreationMessage newMessage, ImmutableList<MessageAttachment> messageAttachments) throws IOException {
+        MultipartBuilder mixedMultipartBuilder = MultipartBuilder.create(MIXED_SUB_TYPE);
+        Map<Boolean, List<MessageAttachment>> distinguishedAttachments = distinguishAttachments(messageAttachments);
+        List<MessageAttachment> inlineAttachments = Optional.ofNullable(distinguishedAttachments.get(true))
+            .orElse(ImmutableList.of());
+        List<MessageAttachment> normalAttachments = Optional.ofNullable(distinguishedAttachments.get(false))
+            .orElse(ImmutableList.of());
+
+        if (!inlineAttachments.isEmpty()) {
+            MultipartBuilder relatedMultipart = MultipartBuilder.create(RELATED_SUB_TYPE);
+            addBody(newMessage, relatedMultipart);
+            addAttachments(inlineAttachments, relatedMultipart);
```

Comment:
It seems weird to delegate this to anoyher function.

I would prefer a method returning the mime related multipart...